roles/adoption_osp_deploy/tasks/config_files.yml

```diff
@@ -70,22 +70,31 @@
         loop_var: group
         label: "{{ group.key }}"
 
+    - name: Override network version protocol vars to ipv6
+      ansible.builtin.set_fact:
+        network_version: network_v6
+        ip_version: ip_v6
+        dns_version: dns_v6
+        gw_version: gw_v6
+        prefix_length_version: prefix_length_v6
+      when: cifmw_networking_env_definition.networks.ctlplane.network_v6 is defined
+
     - name: Generate DeployedServerPortMap field
       vars:
         _node_instance_net: "{{ cifmw_networking_env_definition.instances[node] }}"
         _key_name: "{{ node }}-ctlplane"
-        _ctlplane_ip: "{{ _node_instance_net.networks.ctlplane.ip_v4 }}"
+        _ctlplane_ip: "{{ _node_instance_net.networks.ctlplane[ip_version|default('ip_v4')] }}"
```

Comment:
Would it read better if we remove the `| default()` and instead always `set_fact` above instead?

Something like:
```
ansible.builtin.set_fact:
  ip_version: >-
    {{ 
       "ip_v6" if cifmw_networking_env_definition.networks.ctlplane.network_v6 is defined 
       else "ip_v4"
    }}
```

Then everyhing below can be:
```
_ctlplane_ip: "{{ _node_instance_net.networks.ctlplane[ip_version] }}"
```
 ... etc.